### PR TITLE
Device: Fix TPM for VMs

### PR DESCRIPTION
**Pull request by tomponline**
Fix regression introduced by https://github.com/canonical/lxd/pull/13320

### CODE REVIEW

**Review by MusicDin — APPROVED** [review]
LGTM, thanks

File: lxd/device/tpm.go [hunk 1]
```diff
@@ -217,34 +221,63 @@ func (d *tpm) startVM() (*deviceConfig.RunConfig, error) {
...
-	// Ensure passed file is closed after swtpm start has returned.
-	defer func() {
-		for _, file := range fdFiles {
-			_ = file.Close()
-		}
-	}()
-
-	shortPath, err := util.ShortenedFilePath(socketPath, &fdFiles)
+	// Trickery to handle paths > 108 chars.
+	socketFileDir, err := os.Open(filepath.Dir(socketPath))
```
Comment (MusicDin) [inline lxd/device/tpm.go:228]: While not a blocker, this comment was unclear to me at first. I realized the "trickery" only when I saw the comment of the `shortenedFilePath` function.

Comment (tomponline) [reply lxd/device/tpm.go:228]: thats a fair point indeed, will update, thanks.

Comment (tomponline) [reply lxd/device/tpm.go:228]: will merge this for now so we can check the tests are passing and will open a follow up PR

**Pull request merged**